File: 1-two-sum/1-two-sum.cs

public class Solution {
    public int[] TwoSum(int[] arr, int target)
    {
       
        //target = arr[i] + N
        // N = target - arr[i]
        
       Dictionary<int,int> map = new Dictionary<int,int>();
        
        for(int i=0;i < arr.Length;i++)
        {
            int N = target - arr[i];
            
            if(map.ContainsKey(N))
               return new int[]{map[N],i};
            else if(!map.ContainsKey(arr[i]))
               map.Add(arr[i],i);
        }
        
        return new int[]{-1,-1};
        
        
    }
}

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> map = new HashMap();
                
        for(int i=0;i < nums.length;i++)
        {            
            int N = target - nums[i];    
         
            if(map.containsKey(N))
            {
                return new int[] {map.get(N), i};
            }
            else if(!map.containsKey(nums[i]))
            {
                map.put(nums[i],i);
            }            
        }
        
        return new int[] {-1,-1};        
        
    }
}